src/nodes/NodeBase.ts

```diff
@@ -3,25 +3,25 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-// tslint:disable-next-line:no-require-imports
-import WebSiteManagementClient = require('azure-arm-website');
 import * as path from 'path';
 import * as vscode from 'vscode';
-import { TreeItem } from 'vscode';
+import { Memento, OutputChannel, TreeItem } from 'vscode';
 
 export class NodeBase implements TreeItem {
     public static readonly contextValue: string = 'azureFunctionsNode';
     public contextValue: string;
     public label: string;
     public id: string;
     public collapsibleState: vscode.TreeItemCollapsibleState;
-    public parent: NodeBase;
+    public readonly parent: NodeBase | undefined;
     public command?: vscode.Command;
     public childType?: string;
+    public readonly createChild?: CreateChildFunction;
 
     private children: NodeBase[] | undefined;
 
-    constructor(id: string, label: string, contextValue?: string, commandId?: string) {
+    constructor(parent: NodeBase | undefined, id: string, label: string, contextValue?: string, commandId?: string) {
```

Comment:
Can't we just make this an optional parameter at the end?

Reply:
I intentionally put it at the front because it's an easy thing to forget and theoretically the only nodes we're adding in the future are child nodes (aka 'parent' is essentially required going forward)